### ISSUE TITLE
Support of SOCK5 password authentication for sslocal

### ISSUE THREAD

**Issue by vt-alt** [shadowsocks/shadowsocks-rust#788]
Can you support [SOCK5 authentication](https://datatracker.ietf.org/doc/html/rfc1929) for incoming connections? This may extend use cases.

**Comment by zonyitoo** [comment]
Why is that useful in shadowsocks' use case?

**Comment by vt-alt** [comment]
Selected people could connect from local network using simple socks5 settings, avoiding use by other users from that network.

**Comment by zonyitoo** [comment]
I don't think this is useful because shadowsocks' SOCKS5 UDP port is fixed. SOCKS5's UDP authencation is done by TCP, but shadowsocks wouldn't allocate a new port for every authencated clients.

**Comment by vt-alt** [comment]
What about authentication only for TCP? This still could be useful for http.

**Comment by vt-alt** [comment]
@zonyitoo Thank you very much!
